src/GraphMetadataViewExtension/GraphMetadataViewModel.cs

```diff
@@ -29,13 +29,12 @@ public string GraphDescription
             get { return currentWorkspace.Description; }
             set
             {
-                if (this.currentWorkspace != null && GraphDescription != value)
+                if (currentWorkspace != null && GraphDescription != value)
                 {
-                    this.currentWorkspace.HasUnsavedChanges = true;
+                    MarkCurrentWorkspaceModified();
```

Comment:
This become repeated code in a few places, I would recommend you listen for Model property changed event since you RaisePropertyChanged() any way and you can MarkCurrentWorkspaceModified() in the handler.

Reply:
There is indeed a repetitive pattern across all these properties and we will look to address it as part of the next PRs. Not sure if we want to base that on property changed since we don't want to mark graph as modified every time the underlying property is changed - we want to leave that decision to the caller, in this case, the extension.

Reply:
@BogdanZavu Sure, on the handler you can check the property name and choose what to do. Just seems a bit cleaner and more scalable in the future to me but I will leave the decision to you. example here: https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs#L692